components/compositing/constellation.rs

```diff
@@ -420,104 +411,155 @@ impl<LTF: LayoutTaskFactory, STF: ScriptTaskFactory> Constellation<LTF, STF> {
     }
 
     /// Handles loading pages, navigation, and granting access to the compositor
-    fn handle_request(&mut self, request: ConstellationMsg) -> bool {
+    #[allow(unsafe_code)]
```

Comment:
That's just how the [macro's being expanded](http://doc.servo.org/std/macro.select!.html), and macros are very poor when it comes to throwing helpful errors. I'm not sure why `rustc` behaved that way for you, but removing the unsafe attribute that would otherwise be required by a macro, always throws such an error.